src/transforms/sample/config.rs

```diff
@@ -29,7 +29,16 @@ pub struct SampleConfig {
     /// For example, `rate = 1500` means 1 out of every 1500 events are forwarded and the rest are
     /// dropped.
     #[configurable(metadata(docs::examples = 1500))]
-    pub rate: u64,
+    #[configurable(deprecated = "This option will be deprecated, prefer percent_rate instead")]
+    pub rate: Option<u64>,
+
+    /// The rate at which events are forwarded, expressed as a percentage
+    ///
+    /// For example, `ratio = .13` means that 13% out of all events on the stream are forwarded and
+    /// the rest are dropped.
+    #[configurable(metadata(docs::examples = "0.13"))]
+    #[configurable(validation(range(min = 0.0, max = 1.0)))]
+    pub percent_rate: f32,
```

Comment:
This causes the new `percent_rate` field to be a required setting, breaking all existing configs. It must instead be optional, and the config handling will do a `match (rate, percent_rate)` to check all the outcomes.